BXMSDOC-6481: Release notes docs for DM and PAM 7.9 (#95)
kaldesai/kie-docs: doc-content/enterprise-only/release-notes/rn-790-fixed-issues-ref.adoc

[id='rn-790-fixed-issues-ref']
= Fixed issues in {PRODUCT} {ENTERPRISE_VERSION_LONG}

{PRODUCT} {ENTERPRISE_VERSION_LONG} provides increased stability and fixed issues listed in this section.

== {CENTRAL}

* You cannot save a file when you try to rename the package from the source code [https://issues.redhat.com/browse/RHPAM-3119[RHPAM-3119]]
* The wait spinner is not present when you add a deployment unit, manually enter GAV values, and confirm the alert window [https://issues.redhat.com/browse/RHPAM-3114[RHPAM-3114]]
* Assets are not properly indexed for clustered {CENTRAL} environment [https://issues.redhat.com/browse/RHPAM-3089[RHPAM-3089]]
* When you try to add a new deployment unit, you receive multiple error notifications [https://issues.redhat.com/browse/RHPAM-3112[RHPAM-3112]]
* {CENTRAL} does not remember the last used branch when you log out [https://issues.redhat.com/browse/RHPAM-2957[RHPAM-2957]]

> * In the Assets list, there are columns like `Last modified X day ago` and `Created XX weeks ago` but they are the same for all the assets and does not reflect the actual modified or created values [https://issues.redhat.com/browse/RHPAM-2865[RHPAM-2865]]

Reviewer: ```suggestion
* In the *Assets* list, there are columns like `Last modified X day ago` and `Created XX weeks ago` but they are the same for all the assets and does not reflect the actual modified or created values [https://issues.redhat.com/browse/RHPAM-2865[RHPAM-2865]]
```

* Package list is not populating correctly for imported projects [https://issues.redhat.com/browse/RHDM-1377[RHDM-1377]]

> * `java.lang.Number` import remains in the list of imports even after deletion [https://issues.redhat.com/browse/RHDM-247[RHDM-247]]

Reviewer: ```suggestion
* `java.lang.Number` import remains in the list of imports when deleted [https://issues.redhat.com/browse/RHDM-247[RHDM-247]]
```

> * If the project contains DRL files with the queries, the test scenarios designer displays queries in the coverage report [https://issues.redhat.com/browse/RHDM-1448[RHDM-1448]]

Reviewer: ```suggestion
* If a project contains DRL files with the queries, the test scenarios designer displays queries in the coverage report [https://issues.redhat.com/browse/RHDM-1448[RHDM-1448]]
```

ifdef::PAM[]
* An error message about missing {KIE_SERVER} configuration on dashbuilder runtime must be improved [https://issues.redhat.com/browse/RHPAM-3058[RHPAM-3058]]
* When you remove parent process instance details from the process list, you receive an error message [https://issues.redhat.com/browse/RHPAM-3102[RHPAM-3102]]
* When you skip a task from task event listener, you receive `BAMTaskEventListener` warning [https://issues.redhat.com/browse/RHPAM-3100[RHPAM-3100]]
* To avoid large memory retention, make the `LRUCache` file configurable [https://issues.redhat.com/browse/RHPAM-2808[RHPAM-2808]]
* When more than one {KIE_SERVERS} are connected with different IDs, you receive an unexpected system error [https://issues.redhat.com/browse/RHPAM-3134[RHPAM-3134]]
endif::[]

ifdef::PAM[]

== Process designer

* Case management properties are missing in a case definition [https://issues.redhat.com/browse/RHPAM-3131[RHPAM-3131]]
* If you use the `^` character in the subject of a user task notification an error occurs [https://issues.redhat.com/browse/RHPAM-2763[RHPAM-2763]]

> * Called element in reusable sub process is not populated [https://issues.redhat.com/browse/RHPAM-2760[RHPAM-2760]]

Reviewer: ```suggestion
* Called element in a reusable subprocess is not populated [https://issues.redhat.com/browse/RHPAM-2760[RHPAM-2760]]
```

* Ruleflow group is not populated [https://issues.redhat.com/browse/RHPAM-2740[RHPAM-2740]]

> * If you migrate a process from the legacy process designer to the new process designer, you receive an incorrect warning that a node will be ignored [https://issues.redhat.com/browse/RHPAM-2452[RHPAM-2452]]

Reviewer: ```suggestion
* If you migrate a process from the legacy process designer to the new process designer, you receive an incorrect warning about node being ignored [https://issues.redhat.com/browse/RHPAM-2452[RHPAM-2452]]
```

* In the process modeler, the *Service Tasks* name is changed to *Custom Tasks* [https://issues.redhat.com/browse/RHPAM-3008[RHPAM-3008]]
* The leap days are not validated in timer events [https://issues.redhat.com/browse/RHPAM-2987[RHPAM-2987]]

> * When try you implement `catch-all` errors using an event sub-process, if the value of `ErrorRef` for the error start event is empty, you receive build errors [https://issues.redhat.com/browse/RHPAM-3093[RHPAM-3093]]

Reviewer: ```suggestion
* When try you implement `catch-all` errors using an event subprocess, and if the value of `ErrorRef` for the error start event is empty, you receive build errors [https://issues.redhat.com/browse/RHPAM-3093[RHPAM-3093]]
```

== Process engine

* When you implement the interface, EventEmitter fails to execute the methods [https://issues.redhat.com/browse/RHPAM-3126[RHPAM-3126]]
* When you upgrade {PRODUCT} from 7.7.0 to 7.8.0, the `userTaskService.saveContent` does not work [https://issues.redhat.com/browse/RHPAM-3123[RHPAM-3123]]

> * You can now execute the process with multiple Sources (throw link event) with the same link name and you can use only one Target (catch link event) for the same link name across the sub-process level [https://issues.redhat.com/browse/RHPAM-3132[RHPAM-3132]]

Reviewer: ```suggestion
* You can execute the process with multiple Sources (throw link event) with the same link name and you can use only one Target (catch link event) for the same link name across the subprocess level [https://issues.redhat.com/browse/RHPAM-3132[RHPAM-3132]]
```

* Define jBPM datasource as `@Primary` to avoid the failure of `entityManagerFactory` in `JBPMAutoConfiguration` [https://issues.redhat.com/browse/RHPAM-3072[RHPAM-3072]]

> * A business process with `required` variable tag should not be accepted via REST API [https://issues.redhat.com/browse/RHPAM-3048[RHPAM-3048]]

Reviewer: ```suggestion
* A business process with `required` variable tag should not be accepted using REST API [https://issues.redhat.com/browse/RHPAM-3048[RHPAM-3048]]
```

> * A main process fails due to Sub-process exception handling transaction rollback [https://issues.redhat.com/browse/RHPAM-2982[RHPAM-2982]]

Reviewer: ```suggestion
* The main process fails due to subprocess exception handling transaction rollback [https://issues.redhat.com/browse/RHPAM-2982[RHPAM-2982]]
```

> * When you dynamically add a new node to process instance, the execution of the process fails with a runtime exception [https://issues.redhat.com/browse/RHPAM-3188[RHPAM-3188]]

Reviewer: ```suggestion
* When you dynamically add a new node to a process instance, the execution of the process fails with a runtime exception [https://issues.redhat.com/browse/RHPAM-3188[RHPAM-3188]]
```

endif::[]

== Decision engine

* In decision tables, you can terminate the pattern with `@watch` annotation, which is used to customize the properties that the pattern is reactive on [https://issues.redhat.com/browse/RHDM-1437[RHDM-1437]]
* The `@watch` annotation does not behave properly when you exclude all the properties but one property [https://issues.redhat.com/browse/RHDM-1436[RHDM-1436]]
* Constraint jitting fails When you take a string value from a map and compare it with string literal [https://issues.redhat.com/browse/RHDM-1439[RHDM-1439]]
* If the `accumulate` contains more than one accumulating function, you receive an unwanted re-firing of a rule after deserialization [https://issues.redhat.com/browse/RHDM-1434[RHDM-1434]]

> * Building rules in default-build target of `kie-maven-plugin` takes too much time [https://issues.redhat.com/browse/RHDM-1433[RHDM-1433]]

Reviewer: ```suggestion
* Building rules in a default-build target of `kie-maven-plugin` takes a lot of time [https://issues.redhat.com/browse/RHDM-1433[RHDM-1433]]
```

> * In mvel dialect mode, you cannot compile binary operation for `BigDecimal` type values, you receive a compilation error in executable model [https://issues.redhat.com/browse/RHDM-1421[RHDM-1421]]

Reviewer: ```suggestion
* In MVEL dialect mode, you cannot compile binary operation for `BigDecimal` type values, and you receive a compilation error in the executable model [https://issues.redhat.com/browse/RHDM-1421[RHDM-1421]]
```

== {KIE_SERVER}

* MarshallingException occurs during REST request (JSON) unmarshalling in {KIE_SERVER} [https://issues.redhat.com/browse/RHDM-1417[RHDM-1417]]

== Installer

* On {EAP}, the {PRODUCT} installer adds an additional `$` sign in the vaulted password of database user [https://issues.redhat.com/browse/RHPAM-3148[RHPAM-3148]]

== {OPENSHIFT}

* In monitoring console page, when you create a new page, you receive an *Activity not found* error pop-up [https://issues.redhat.com/browse/RHPAM-3066[RHPAM-3066]]

ifdef::PAM[]
* When you try to configure an external {RH-SSO} configuration, you receive `SSL required for: EXTERNAL` error message [https://issues.redhat.com/browse/RHPAM-2804[RHPAM-2804]]
* Missing support for `xa-pool` configuration of `is-same-rm-override` field [https://issues.redhat.com/browse/RHPAM-3071[RHPAM-3071]]

> * In {OPENSHIFT}, when you create a secure connection between {CENTRAL} and {KIE_SERVER}, a wrong URL is generated [https://issues.redhat.com/browse/RHPAM-3162[RHPAM-3162]]

Reviewer: ```suggestion
* In {OPENSHIFT}, when you create a secure connection between {CENTRAL} and {KIE_SERVER}, an incorrect URL is generated [https://issues.redhat.com/browse/RHPAM-3162[RHPAM-3162]]
```
